bin/packages/build.js

```diff
@@ -18,6 +18,52 @@ const files = process.argv.slice( 2 );
  */
 const PACKAGES_DIR = path.resolve( __dirname, '../../packages' );
 
+/**
+ * Get the package name for a specified file
+ *
+ * @param  {string} file File name
+ * @return {string}      Package name
+ */
+function getPackageName( file ) {
+	return path.relative( PACKAGES_DIR, file ).split( path.sep )[ 0 ];
+}
+
+/**
+ * Returns a stream transform which maps an individual stylesheet to its
+ * package entrypoint. Unlike JavaScript which uses an external bundler to
+ * efficiently manage rebuilds by entrypoints, stylesheets are rebuilt fresh
+ * in their entirety from the build script.
+ *
+ * @return {Transform} Stream transform instance.
+ */
+function createStyleEntryTransform() {
+	const packages = new Set;
+
+	return new Transform( {
+		objectMode: true,
+		async transform( file, encoding, callback ) {
+			// Only stylesheets are subject to this transform.
```

Comment:
I thought to be clever / functional here with some sort of conditional flow (an [`overEvery`](https://lodash.com/docs/4.17.11#overEvery) and/or series `Promise`) but I couldn't come up with an implementation which wouldn't end up being unnecessarily obscure / unreadable.

Reply:
Best to keep it simple! 😀